src/main/java/com/example/petbuddybackend/config/websocket/SubscriptionInterceptor.java

```diff
@@ -0,0 +1,49 @@
+package com.example.petbuddybackend.config.websocket;
+
+import com.example.petbuddybackend.service.chat.ChatService;
+import com.example.petbuddybackend.utils.exception.throweable.chat.NotParticipateException;
+import lombok.RequiredArgsConstructor;
+import org.springframework.messaging.Message;
+import org.springframework.messaging.MessageChannel;
+import org.springframework.messaging.simp.stomp.StompCommand;
+import org.springframework.messaging.simp.stomp.StompHeaderAccessor;
+import org.springframework.messaging.support.ChannelInterceptor;
+import org.springframework.stereotype.Component;
+
+import java.security.Principal;
+
+@Component
+@RequiredArgsConstructor
+public class SubscriptionInterceptor implements ChannelInterceptor {
+
+    private final ChatService chatService;
+
+    @Override
+    public Message<?> preSend(Message<?> message, MessageChannel channel) {
```

Comment:
a to sie wywola przed wyslaniem wiadomości, czy przed dolaczeniem do kanalu?

Reply:
To się zawsze wywołuje jak się robi coś z websocketami (połączenie, subskrypcje, rozłączenie, wysłanie wiadomości). Niżej jest `StompCommand.SUBSCRIBE` żeby filtrować i zajmować się tylko eventami związanymi z subskrypcją na topic

Zaś wysyłanie wiadomości zabezpieczam tym `@PreAuthorize("isAuthenticated()")`